File: C/16_Greedy/BOJ_02_1931_회의실배정.c
#include <stdio.h>
#include <stdlib.h>
int n, time[100000][2];
int compare(const void* a, const void* b) {
	int* o1 = (int*)a;
	int* o2 = (int*)b;
	if (o1[1] == o2[1]) {
		return o1[0] - o2[0];
	}
	return o1[1] - o2[1];
}
main() {
	scanf("%d", &n);
	for (int i = 0; i < n; i++) {
		for (int j = 0; j < 2; j++) {
			scanf("%d", &time[i][j]);
		}
	}
	qsort(time, n, sizeof(int) * 2, compare);
	int cnt = 0, end = 0;
	for (int i = 0; i < n; i++) {
		if (end <= time[i][0]) {
			end = time[i][1];
			cnt++;
		}
	}
	printf("%d", cnt);
}

File: JAVA/16_Greedy/BOJ_02_1931_회의실배정.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.Comparator;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[][] time = new int[n][2];
        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine(), " ");
            for (int j = 0; j < 2; j++) {
                time[i][j] = Integer.parseInt(st.nextToken());
            }
        }
        Arrays.sort(time, new Comparator<int[]>() {
            @Override
            public int compare(int[] o1, int[] o2) {
                if (o1[1] == o2[1]) {
                    return o1[0] - o2[0];
                }
                return o1[1] - o2[1];
            }
        });
        int cnt = 0, end = 0;
        for (int i = 0; i < n; i++) {
            if (end <= time[i][0]) {
                end = time[i][1];
                cnt++;
            }
        }
        System.out.println(cnt);
    }
}